scripts/hetnet_id_extractor.py

```diff
@@ -1,34 +1,59 @@
-    disease_merged_df = disease_merged_df.drop_duplicates()
-    disease_merged_df = disease_merged_df[["pubmed_id", "type", "offset", "end", "doid_code"]].rename(columns={"doid_code": "identifier"})
-
-    # Verify Gene IDs are human genes
-    gene_df = extracted_tag_df[extracted_tag_df["type"] == "Gene"]
-    gene_final_df = gene_df[gene_df["identifier"].isin(hetnet_gene_df["GeneID"])]
-
-    final_df = gene_final_df
-    final_df = final_df.append(chemical_merged_df)
-    final_df = final_df.append(disease_merged_df)
-
-    final_df[["pubmed_id", "type", "identifier", "offset", "end"]].sort_values(["pubmed_id", "offset"]).to_csv(args.output, sep="\t", index=False)
+    """ This method filters pubtator tags to consist of only 
+        hetnet tags
+
+        Keyword arguments:
+        infile -- the name of the file to read
+        outfile -- the name of the output file
+    """
+
+    print_header = True
+    for extracted_tag_df in tqdm.tqdm(get_tag_chunks(infile)):
+        hetnet_chemical_df = load_chemical_df()
+        hetnet_disease_df = load_disease_df()
+        hetnet_gene_df = load_gene_df()
+
+        # Covert chemical IDs
+        chemical_merged_df = pd.merge(extracted_tag_df[extracted_tag_df["type"] == "Chemical"], hetnet_chemical_df[["drugbank_id", "identifier"]], left_on="identifier", right_on="identifier")
+        chemical_merged_df = chemical_merged_df.drop_duplicates()
+        chemical_merged_df["type"] = "Compound"
+        chemical_merged_df = chemical_merged_df[["pubmed_id", "type", "offset", "end", "drugbank_id"]].rename(columns={"drugbank_id": "identifier"})
+
+        # Convert Disease IDs
+        disease_merged_df = pd.merge(extracted_tag_df[extracted_tag_df["type"] == "Disease"], hetnet_disease_df[["doid_code", "resource_id"]], left_on="identifier", right_on="resource_id")
+        disease_merged_df = disease_merged_df.drop_duplicates()
+        disease_merged_df = disease_merged_df[["pubmed_id", "type", "offset", "end", "doid_code"]].rename(columns={"doid_code": "identifier"})
+
+        # Verify Gene IDs are human genes
+        gene_df = extracted_tag_df[extracted_tag_df["type"] == "Gene"]
+        gene_final_df = gene_df[gene_df["identifier"].isin(hetnet_gene_df["GeneID"])]
+
+        final_df = gene_final_df
+        final_df = final_df.append(chemical_merged_df)
+        final_df = final_df.append(disease_merged_df)
+
+        if print_header:
+            final_df[["pubmed_id", "type", "identifier", "offset", "end"]].sort_values(["pubmed_id", "offset"]).to_csv(args.output, sep="\t", index=False, mode='a', compression="xz")
+            print_header = False
+        else:
+            final_df[["pubmed_id", "type", "identifier", "offset", "end"]].sort_values(["pubmed_id", "offset"]).to_csv(args.output, sep="\t", index=False, mode='a', header=False, compression="xz")
```

Comment:
This will always xz compress, even when filename does not end in `.xz`.

For example, did you rerun `tests.sh` and track the changes of  `data/example/4-hetnet-tags.tsv`?

I think you can use `get_opener` from utilities and then use `'at'` mode for append text to open a file that you pass to pandas.DataFrame.to_csv().